pkg/engine/deployment.go

```diff
@@ -427,8 +429,8 @@ func isDefaultProviderStep(step deploy.Step) bool {
 	return providers.IsDefaultProvider(step.URN())
 }
 
-func checkTargets(targetUrns deploy.UrnTargets, snap *deploy.Snapshot) error {
-	if !targetUrns.IsConstrained() {
+func checkTargets(targetUrns deploy.UrnTargets, excludeUrns deploy.UrnTargets, snap *deploy.Snapshot) error {
+	if !targetUrns.IsConstrained() || !excludeUrns.IsConstrained() {
```

Comment:
Should this `||` be an `&&`? Right now it reads "if the set of target URNs is not constrained, or the set of exclude URNs is not constrained", which feels a bit weird to me, but maybe works because of mutual exclusion/something else I'm not considering?